drivers/include/nrf24l01p.h

```diff
@@ -1,627 +1,1366 @@
-
-/**
-* @brief   Disable RX datapipe on the nrf24l01+ transceiver.
-*
-* @param[in] dev    Transceiver device to use.
-* @param[in] pipe   RX pipe to disable.
-*
-* @return           0 on success.
-* @return           -1 on error.
-*/
-int nrf24l01p_disable_pipe(const nrf24l01p_t *dev, nrf24l01p_rx_pipe_t pipe);
-
-/**
-* @brief   Disable CRC error detection on the nrf24l01+ transceiver.
-*
-* @param[in] dev    Transceiver device to use.
-*
-* @return           0.
-*/
-int nrf24l01p_disable_crc(const nrf24l01p_t *dev);
-
-/**
-* @brief   Enable CRC error detection on the nrf24l01+ transceiver.
-*
-* @param[in] dev    Transceiver device to use.
-* @param[in] crc    Length of cyclic redundancy check (type nrf24l01p_crc_t).
-*
-* @return           0 on success.
-* @return           -1 on error.
-*/
-int nrf24l01p_enable_crc(const nrf24l01p_t *dev, nrf24l01p_crc_t crc);
-
-/**
-* @brief   Setup and enable automatic ACK and retransmission on the nrf24l01+ transceiver.
-*
-* @note
-* This function enables automatic acknowledgement for a given RX data pipe and also sets up the
-* mautomatic retransmission behavior.
-*
-* @param[in] dev    Transceiver device to use.
-* @param[in] pipe   RX pipe to setup auto ack.
-* @param[in] delay_retrans    Automatic retransmission delay
-                              (type nrf24l01p_retransmit_delay_t)
-* @param[in] count_retrans    Auto retransmit count.
-*
-* @return           0 on success.
-* @return           -1 on error.
-*/
-int nrf24l01p_setup_auto_ack(const nrf24l01p_t *dev, nrf24l01p_rx_pipe_t pipe, nrf24l01p_retransmit_delay_t delay_retrans, char count_retrans);
-
-/**
-* @brief   Disable automatic ACK on the nrf24l01+ transceiver.
-*
-* @param[in] dev    Transceiver device to use.
-*
-* @return           0 on success.
-* @return           -1 on error.
-*/
-int nrf24l01p_disable_all_auto_ack(const nrf24l01p_t *dev);
-
-/**
-* @brief   Flush TX FIFO on the nrf24l01+ transceiver.
-*
-* @param[in] dev    Transceiver device to use.
-*
-* @return           0 on success.
-* @return           -1 on error.
-*/
-int nrf24l01p_flush_tx_fifo(const nrf24l01p_t *dev);
-
-/**
-* @brief   Flush RX FIFO on the nrf24l01+ transceiver.
-*
-* @param[in] dev    Transceiver device to use.
-*
-* @return           0 on success.
-* @return           -1 on error.
-*/
-int nrf24l01p_flush_rx_fifo(const nrf24l01p_t *dev);
-
-/**
-* @brief   Callback that is called when interrupt occurs on interrupt
-* pin from the nrf24l01+ transceiver.
-*
-* @param[in] arg    Used to pass transceiver device "dev".
-*/
-void nrf24l01p_rx_cb(void *arg);
+    NRF24L01P_RF_PWR_MINUS_18dBm    = 0,    /**< -18 dBm */
+    NRF24L01P_RF_PWR_MINUS_12dBm    = 1,    /**< -12 dBm */
+    NRF24L01P_RF_PWR_MINUS_6dBm     = 2,    /**<  -6 dBm */
+    NRF24L01P_RF_PWR_0dBm           = 3,    /**<   0 dBm */
+    NRF24L01P_RF_PWR_NUM_OF                 /**< Number of possible values to configure the radio power */
+} nrf24l01p_rfpwr_t;
```

Comment:
Maybe for consistency of names in RIOT this should be renamed to `NRF24L01P_TX_POWER_MINUX_18DBM`? And `nrf24l01p_rfpwr_t` could be `nrf24l01p_tx_power_t`.

Also: Please avoid camel case (`dBm`).

Reply:
Ok